rust/xaynet-server/src/storage/store.rs

```diff
@@ -0,0 +1,143 @@
+//! A generic store.
+
+use crate::{
+    state_machine::coordinator::CoordinatorState,
+    storage::{
+        CoordinatorStorage,
+        LocalSeedDictAdd,
+        MaskScoreIncr,
+        ModelStorage,
+        StorageResult,
+        SumPartAdd,
+    },
+};
+use xaynet_core::{
+    common::RoundSeed,
+    mask::{MaskObject, Model},
+    LocalSeedDict,
+    SeedDict,
+    SumDict,
+    SumParticipantEphemeralPublicKey,
+    SumParticipantPublicKey,
+    UpdateParticipantPublicKey,
+};
+
+#[derive(Clone)]
+/// A generic store.
+pub struct Store<C, M>
+where
+    C: CoordinatorStorage,
+    M: ModelStorage,
+{
+    /// A coordinator store.
+    coordinator: C,
+    /// A model store.
+    model: M,
+}
+
+impl<C, M> Store<C, M>
```

Comment:
reading this impl block i was wondering why you didn't impl the individual traits to expose their methods?

like having a `impl<C, M> Store<C, M> {}` for the `new()` method and then
```rust
impl<C, M> CoordinatorStorage for Store<C, M> {}
impl<C, M> ModelStorage for Store<C, M> {}
```

this way no method of the traits might be accidentally forgotten and changes to the trait will immediately be noticed in the `Store` impl. this is not to be confused with an overarching storage trait like in option 2 and 3 of #593.

Reply:
I didn't do it because otherwise the method `create_global_model_id` would have to be called in every concrete implementation that implements the` ModelStorage` trait.

https://github.com/xaynetwork/xaynet/blob/d1cfc811917ac21d196a7e4dbf35030234c09213/rust/xaynet-server/src/storage/store.rs#L127-L137

I am not sure what is the best solution here. what do you prefer?

Reply:
mhh, i see, so it seems to me that `create_global_model_id` is a helper method that we use for the `ModelStorage` trait impl and we don't expect every implementer of the trait to actually implement or use this, right? then it probably shouldn't be part of the trait itself, but instead it could be a helper function either living in the `api` module or the `store` module or some utils module.

Reply:
I implemented the `CoordinatorStorage` and the `ModelStorage` trait for `Store`. Currently I prefer to keep the `create_global_model_id` method in the `ModelStorage` trait. Maybe I will make it a helper function later. Let's see how it will look like once the store is integrated.